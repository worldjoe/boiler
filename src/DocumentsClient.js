class DocumentsClient {

  async getDocument(id) {
    const response = await fetch("https://localhost:8443/v1/documents/" + id, { method: 'GET'});
    const documentResponse = await response.json();
    return documentResponse;
  }

  async createDocument(content) {
    const response = await fetch("https://localhost:8443/v1/documents/", {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({'content' : content}),
    });
    const documentResponse = await response.json();
    return documentResponse;
  }

  async deleteDocument(id) {
    const response = await fetch("https://localhost:8443/v1/documents/" + id, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    //    const documentResponse = await response.json();
    return response;
  }

  async getAllDocuments() {
    console.log("joey in getAllDocuments");
    const response = await fetch("https://localhost:8443/v1/documents/", { method: 'GET'});
    console.log("joey in getAllDocuments response");
    const documentResponse = await response.json();
    console.log("joey in getAllDocuments response getting json");
    return documentResponse;
  }

  async updateDocument(id, content) {
    const response = await fetch("https://localhost:8443/v1/documents/" + id, {
      method: 'PUT', // or 'POST'
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({'content' : content}),
    });
    const documentResponse = await response.json();
    return documentResponse;
  }
}
export default DocumentsClient;
