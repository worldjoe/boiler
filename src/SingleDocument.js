import React, { Component } from "react";
import DocumentsClient from './DocumentsClient.js';
import { connect } from 'react-redux';
import { addDocumentAsync, deleteDocumentAsync, editDocumentAsync } from './redux/actions';

class SingleDocument extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    //    console.log("Joey props");
    //console.log(this.props.document.id);
    this.state = (props.document) ? props.document : { 'content': ''};
    //this.state = props;
  }

  handleChange = event => {
    this.setState({ content: event.target.value });
  };

  handleAddDocument = () => {
    this.props.addDocumentAsync(this.state.content);
    this.setState({ content : "" });
  };

  handleDeleteDocument = () => {
    this.props.deleteDocumentAsync(this.state.id);
    this.setState({ content : "" });
  };

  handleEditDocument = () => {
    this.props.editDocumentAsync(this.state.id, this.state.content);
    this.setState({ content : this.state.content });
  };

  render() {
  return (
    <div className="SingleDocument" id={this.state.id}>
      <form>
        {this.state.id ? (
          <React.Fragment>
        <label>id</label>
        <input readOnly value={this.state.id}/></React.Fragment>) : <div/>}

        <label>content</label>
        <input value={this.state.content} onChange={this.handleChange}/>
        <input type='button'  onClick={this.state.id ? this.handleEditDocument : this.handleAddDocument } value={this.state.id ? ('submit') : 'new'}/>
        {this.state.id ? (
          <input type='button'  onClick={this.handleDeleteDocument} value='delete'/>
        ) : <div/>}
      </form>
    </div>
  );
  }
}

export default connect(null, {addDocumentAsync, deleteDocumentAsync, editDocumentAsync} ) (SingleDocument);
