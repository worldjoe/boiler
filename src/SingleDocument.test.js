/* Dependencies */
import React from 'react'
import renderer from 'react-test-renderer';
import SingleDocument from './SingleDocument.js'
import * as DocumentsClient from './DocumentsClient.js';
jest.mock('./DocumentsClient');

describe('test window location\'s reload function', () => {
  const location = window.location;
  delete window.location;
  window.location = {
    ...location,
    reload: jest.fn()
  };;
  test ("donothing", () => {

  });
/*
  test("testRender", () => {
    //  let documentsClient = new DocumentsClient();
    let mockReRenderCallbackFunction = jest.fn();
    const singleDocumentComponent = renderer.create(
      <SingleDocument rerenderParentCallback={mockReRenderCallbackFunction} />,
    );
    let tree = singleDocumentComponent.toJSON();
    expect(tree).toMatchSnapshot();
    singleDocumentComponent.root.findAllByType('input')[1].props.onClick();
    return new Promise(resolve => setImmediate(resolve)).then (() => {
      console.log("Joey in promise return going to check window.location.reload");

      expect(mockReRenderCallbackFunction).toHaveBeenCalled();
      //expect(window.location.reload).toHaveBeenCalledTimes(1);
      jest.restoreAllMocks();
      window.location = location;
    });
  });

  test("testRender with content", () => {
    let props = {'content': 'joey is awesome', 'id' :'123'};
    //  let documentsClient = new DocumentsClient();
    let mockReRenderCallbackFunction = jest.fn();
    const singleDocumentComponent = renderer.create(
      <SingleDocument document={props} rerenderParentCallback={mockReRenderCallbackFunction} />,
    );
    let tree = singleDocumentComponent.toJSON();
    expect(tree).toMatchSnapshot();
    singleDocumentComponent.root.findAllByType('input')[2].props.onClick();
    return new Promise(resolve => setImmediate(resolve)).then (() => {
      console.log("Joey in promise return going to check window.location.reload");

      jest.restoreAllMocks();
      window.location = location;
    });
  });
  */
});