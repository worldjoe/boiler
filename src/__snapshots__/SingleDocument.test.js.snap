// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test window location's reload function testRender 1`] = `
<div
  className="SingleDocument"
>
  <form>
    <div />
    <label>
      content
    </label>
    <input
      onChange={[Function]}
      value=""
    />
    <input
      onClick={[Function]}
      type="button"
      value="new"
    />
    <div />
  </form>
</div>
`;

exports[`test window location's reload function testRender with content 1`] = `
<div
  className="SingleDocument"
  id="123"
>
  <form>
    <label>
      id
    </label>
    <input
      readOnly={true}
      value="123"
    />
    <label>
      content
    </label>
    <input
      onChange={[Function]}
      value="joey is awesome"
    />
    <input
      onClick={[Function]}
      type="button"
      value="submit"
    />
    <input
      onClick={[Function]}
      type="button"
      value="delete"
    />
  </form>
</div>
`;
